src/interpreter/mod.rs

```diff
@@ -243,8 +243,11 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
 
             Str(ref s) => {
                 let ptr = self.memory.allocate(s.len(), 1)?;
-                self.memory.write_bytes(ptr, s.as_bytes())?;
-                self.memory.freeze(ptr.alloc_id)?;
+                // don't write to the zst allocation
```

Comment:
writing zero bytes will still call `get_mut` on the allocation. I could special case `write_bytes` and friends to ignore arguments with zero bytes.